content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,123 @@
+
+Every extended client operates independently, in an isolated instance. It cannot conflict with other extended clients or the original client.
+
+You can also combine two or more extensions into one extension, if you want the functionality from both extensions to apply in the same client instance. Note that it is possible for [combined extensions to conflict](#conflicts-in-combined-extensions).
+
+### Examples of isolated client instances
+
+Example uses of isolated client instances include the following:
+
+- Implement row-level security (RLS), where each http request has its own client with its own RLS extension, customized with session data. This can keep each user entirely separate, each in a separate client.
+- Add a `user.current()` method for the `user` model to get the current logged-in user
+- Enable more verbose logging for requests if a debug cookie is set
+- Attach a unique request id to all logs so you can correlate them later
+- Remove a `delete` method from models unless the application calls the admin endpoint and the user has the necessary priviliges
+
+### Declare multiple extensions
+
+The following example declares two separate extensions and one combined extension:
+
+```ts
+// First of all, store your original prisma client in a variable as usual
+const prisma = new PrismaClient()
+
+// Declare an extended client that has extension A applied
+const extendsA = prisma.$extends(extensionA)
+
+// Declare an extended client that has extension B applied
+const extendsB = prisma.$extends(extensionB)
+
+// Declare an extended client that has extensions A and B applied
+const extendsAB = prisma.$extends(extensionA).$extends(extensionB)
+
+//Declare an extended client that has a stacked combination of extensions A and B
+const extendsAplusB = extendsA.$extends(extensionB)
+```
+
+### Conflicts in combined extensions
+
+When you combine two or more extensions, then the _last_ extension that you declare takes precedence in any conflict. In the example above, suppose there is a method called `getName()` in `extendsA` and a method called `getName()` in `extendsB`. When you call `extendsAB.getName()`, then the client uses `getName()` as defined in `extendsB`.
+
+## Extensions versus middlewares
+
+Extensions differ from [middlewares](/concepts/components/prisma-client/middleware#performance-and-appropriate-use-cases) in the following ways:
+
+- If you use two or more middlewares at the same time, then they all apply globally to the same client.
+- Extensions are isolated: the functionality in each one is separate, unless you deliberately combine them.
+  - For example, in a row-level security (RLS) scenario, extensions keep each user in an entirely separate client. With middlewares, all users are active in the same client.
+- During application execution, you can enable and disable extended clients, but you cannot enable or disable middlewares.
```

Comment:
I do not know if "enable and disable" is the correct terminology here.
You can use a forked client with the extension, or the original, vanilla one that does not have the extension applied.

Reply:
OK, I've updated this.